test/Common.cpp

```diff
@@ -207,6 +211,45 @@ bool CommonOptions::parse(int argc, char const* const* argv)
 	return true;
 }
 
+string CommonOptions::toString(vector<string> const& _selectedOptions) const
+{
+	if (!m_singleton || _selectedOptions.empty())
+		return "";
+
+	auto boolToString = [](bool value) -> string
+		{
+			return value ? "true" : "false";
+		};
+	// Using std::map to avoid if-else/switch-case block
+	map<string, string> optionValueMap = {
+		{"evmVersion", "evmVersion=" + evmVersion().name()},
+		{"optimize", "optimize=" + boolToString(optimize)},
+		{"useABIEncoderV1", "useABIEncoderV1=" + boolToString(useABIEncoderV1)},
+		{"batch", "batch=" + std::to_string(selectedBatch + 1) + "/" + std::to_string(batches)},
+		{"ewasm", "ewasm=" + boolToString(ewasm)},
+		{"enforceCompileToEwasm", "enforceCompileToEwasm=" + boolToString(enforceCompileToEwasm)},
+		{"enforceGasTest", "enforceGasTest=" + boolToString(enforceGasTest)},
+		{"enforceGasTestMinValue", "enforceGasTestMinValue=" + enforceGasTestMinValue.str()},
+		{"disableSemanticTests", "disableSemanticTests=" + boolToString(disableSemanticTests)},
+		{"disableSMT", "disableSMT=" + boolToString(disableSMT)},
+		{"showMessages", "showMessages=" + boolToString(showMessages)},
+		{"showMetadata", "showMetadata=" + boolToString(showMetadata)}
+	};
+
+	using namespace ranges;
+	vector<string> options = _selectedOptions |
+		views::transform([&optionValueMap](auto& optionName) { return optionValueMap.find(optionName); }) |
+		views::filter([&optionValueMap](auto it) { return it != optionValueMap.end(); }) |
+		views::transform([](auto it) { return it->second; }) |
+		to<vector>();
```

Comment:
1) I just noticed that you just ignore options that are not present in `optionValueMap`. I think we should require them to be present instead. Option names are not user input so we do not need to handle them gracefully and in practice, when you make a typo in the name it's better to get an error than for it to be silently skipped.
2) Since the `option=` part is always identical to the key, why not remove it from the map and add it here? Less duplication and the map would be more readable.

```suggestion
	soltestAssert(views::set_difference(_selectedOptions, optionValueMap | views::keys).empty());

	vector<string> optionsWithValues = _selectedOptions |
		views::transform([&optionValueMap](string const& _option) { return _option + "=" + optionValueMap.at(_option); }) |
		to<vector>();
```

Reply:
1. Yes, it really makes sense to do so. I did not think about erroneous values at the time I came up with that code.
2. I should have done it like that since the beginning.
Thanks for the excellent review and suggestions!

Reply:
I could not use `set_differences` because `_selectedOptions` should be sorted prior to that, but then it is a `const` reference. So I used `all_of` instead...